src/core/main.c

```diff
@@ -1535,9 +1538,12 @@ static int become_shutdown(
                 watchdog_timer = arg_kexec_watchdog;
 
         if (watchdog_timer > 0 && watchdog_timer != USEC_INFINITY) {
-
+                usec_t watchdog_pretimeout = 0;
                 char *e;
 
+                /* Disable watchdog pretimeouts for shutdown. */
+                watchdog_set_pretimeout(&watchdog_pretimeout);
```

Comment:
the parameter is not actually used, hence make it optional in watchdog_set_pretimeout() so that we can just pass NULL here